modules/vendor/transform.lua

```diff
@@ -1,24 +1,37 @@
 package.path = package.path .. ";../bytes/?.lua;"
-local transform = require("parser_factory")
-
--- Below an example of transforming the uplink message from a device with custom payloads parser
--- help can be found here https://docs-staging.exosite.com/development/services/sigfox/decoding-grammar/
-
-local res1 = {["resource"]= "bool1", ["definition"] = ":bool:2"}
-local res2 = {["resource"]= "uint1", ["definition"] = ":uint:16"}
-local res3 = {["resource"] = "float1", ["definition"] = ":float:32"}
-local res4 = {["resource"]= "int1", ["definition"] = ":int:16"}
-local payloads = 
-{
-  res1, res2, res3, res4
-}
-local response = transform.parse_payloads(payloads,"C123405C82548A3E")
-print(response.bool1) -- false
-print(response.uint1) -- 49443
-print(response.float1) -- -3.4454545974731
-print(response.int1) -- -30146
+local parser_factory = require("parser_factory")
+local transform = {}
+-- Below an example of transforming the uplink message from a device, depending mode activated
+
+function transform.DecodeMode(uplink)
+  --used for ConvertStateUplink, can have more switch case
+  --mode is given on first byte
+  local mode = parser_factory.getuint(parser_factory.fromhex(uplink),0,8)
+  if mode == 1 then
+    -- mode 1 enter temperature as float for ex.
+    return {['Temperature'] = parser_factory.getfloat_32(parser_factory.fromhex(uplink),1)}
+  elseif mode == 2 then
+    -- mode 2 enter device's button state as boolean for ex.
+    return {['Button State'] = parser_factory.getbool(parser_factory.fromhex(uplink),1,2)}    
+  else
+    return {'no Data associated'}
+  end
+end
+
+function transform.ConvertStateUplink(state)
+  --one data_in per state.uplink
+  state.data_in.decoded = json.encode(transform.DecodeMode(state.uplink))
+  return state
+end
 
 
+local state1 = {
+  ["uplink"] = "014091cfab",
+  ["data_in"] = {}
+}
+transform.ConvertStateUplink(state1)
```

Comment:
```suggestion
transform.ConvertStateUplink(state.data_in)
```
And move it to line 64 with a comment explaining

Reply:
> And move it to line 64 with a comment explaining

Do we have to add a second argument with _state.uplink_ ? as it is needed in ConvertStateUplink

Reply:
@sauvainr 
`uplink` is stored in `event.payload` (not `event.payload.state`):
```lua
local uplink = event.payload.uplink
local data_in = event.payload.state.data_in
transform.convert(uplink, data_in)
```
so it would be better defining like this?
input -> uplink
output -> data_in

Reply:
@lenaw I think state here == event.payload